services/provider/main.go

```diff
@@ -13,6 +17,58 @@ var (
 
 func main() {
 	flag.Parse()
-	var opts []grpc.ServerOption
-	server.Start(*port, opts)
+
+	klog.Info("Starting Provider API server")
+
+	namespace, err := util.GetWatchNamespace()
+	if err != nil {
+		klog.Errorf("failed to get provider cluster namespace. Error %v", err)
+		return
+	}
+
+	consumerAPI, err := consumer.NewConsumerAPIUpdater()
+	if err != nil {
+		klog.Errorf("failed to initialize consumer client. Error %v", err)
+		return
+	}
+
+	// Get master key to encryt/decryt token
+	// TODO: Remove hard coded value for secret name
+	secret, err := consumerAPI.GetSecret("ocs-provider-server", namespace)
+	if err != nil {
+		klog.Errorf("failed to get secret to read the master key. Error %v", err)
+		return
+	}
+
+	masterKey := string(secret.Data["Key"])
+	if masterKey == "" {
+		klog.Errorf("master key is empty")
+		return
+	}
+
+	// Update in-memory user store
+	storageConsumers, err := consumerAPI.ListStorageConsumers(namespace)
+	consumerStore := consumer.NewConsumerStore()
+	if err != nil {
+		klog.Errorf("failed to list storageConsumer resources in %q namespace. Error %v", namespace, err)
+	}
+
+	for _, sc := range storageConsumers.Items {
+		newConsumer := consumer.NewConsumer(sc.Name, string(sc.UID), sc.Annotations[common.TokenAnnotation])
+		err := consumerStore.Save(newConsumer)
+		if err != nil {
+			klog.Errorf("failed to intialize in-memory consumer store. Error %v", err)
```

Comment:
```suggestion
			klog.Errorf("failed to initialize in-memory consumer store. Error %v", err)
```

Is the `in-memory` message relevant?

Reply:
maybe failed to initialize local consumer cache? 

As a side note, I think the struct is miss-named and misleading.  
A store usually holds the source of truth for its items while this struct acts more like a cache, so maybe ConsumerCache would be preferable (for both the type and the message)

Reply:
using `cache` instead of `store`.